src/SerializeAST.cc

```diff
@@ -0,0 +1,766 @@
+                break;
+
+            default:
+                break;
+        }
+
+        return sos::String(baseTypeName);
+    }
+
+    return WrapSymbol(typeName.symbol);
+}
+
+
+sos::Object WrapTypeSpecification(const mson::TypeSpecification& typeSpecification)
+{
+    sos::Object typeSpecificationObject;
+
+    // Name
+    typeSpecificationObject.set(SerializeKey::Name, WrapTypeName(typeSpecification.name));
+
+    // Nested Types
+    typeSpecificationObject.set(SerializeKey::NestedTypes, 
+                                WrapCollection<mson::TypeName>()(typeSpecification.nestedTypes, WrapTypeName));
+
+    return typeSpecificationObject;
+}
+
+sos::Array WrapTypeAttributes(const mson::TypeAttributes& typeAttributes)
+{
+    sos::Array typeAttributesArray;
+
+    if (typeAttributes & mson::RequiredTypeAttribute) {
+        typeAttributesArray.push(sos::String("required"));
+    }
+    else if (typeAttributes & mson::OptionalTypeAttribute) {
+        typeAttributesArray.push(sos::String("optional"));
+    }
+    else if (typeAttributes & mson::DefaultTypeAttribute) {
+        typeAttributesArray.push(sos::String("default"));
+    }
+    else if (typeAttributes & mson::SampleTypeAttribute) {
+        typeAttributesArray.push(sos::String("sample"));
+    }
+    else if (typeAttributes & mson::FixedTypeAttribute) {
+        typeAttributesArray.push(sos::String("fixed"));
+    }
+
+    return typeAttributesArray;
+}
+
+sos::Object WrapTypeDefinition(const mson::TypeDefinition& typeDefinition)
+{
+    sos::Object typeDefinitionObject;
+
+    // Type Specification
+    typeDefinitionObject.set(SerializeKey::TypeSpecification, WrapTypeSpecification(typeDefinition.typeSpecification));
+
+    // Type Attributes
+    typeDefinitionObject.set(SerializeKey::Attributes, WrapTypeAttributes(typeDefinition.attributes));
+
+    return typeDefinitionObject;
+}
+
+sos::Object WrapValueDefinition(const mson::ValueDefinition& valueDefinition)
+{
+    sos::Object valueDefinitionObject;
+
+    // Values
+    valueDefinitionObject.set(SerializeKey::Values,
+                              WrapCollection<mson::Value>()(valueDefinition.values, WrapValue));
+
+    // Type Definition
+    valueDefinitionObject.set(SerializeKey::TypeDefinition, WrapTypeDefinition(valueDefinition.typeDefinition));
+
+    return valueDefinitionObject;
+}
+
+sos::Object WrapPropertyName(const mson::PropertyName& propertyName)
+{
+    sos::Object propertyNameObject;
+
+    if (!propertyName.literal.empty()) {
+        propertyNameObject.set(SerializeKey::Literal, sos::String(propertyName.literal));
+    }
+    else if (!propertyName.variable.empty()) {
+        propertyNameObject.set(SerializeKey::Variable, WrapValueDefinition(propertyName.variable));
+    }
+
+    return propertyNameObject;
+}
+
+// Forward declarations
+sos::Object WrapTypeSection(const mson::TypeSection& typeSection);
+
+
+const sos::String TypeSectionToString(const mson::TypeSection& section)
```

Comment:
Yeah, but shouldn't this function be named `TypeSectionClassToString` and take `mson::TypeSection::Class` as input? Why does it need the full type section?